runtime/opt/taupage/init.d/02-register-td-agent.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+
+import logging
+import subprocess
+import re
+import boto.utils
+
+from jinja2 import Environment, FileSystemLoader
+from taupage import get_config
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+TPL_NAME = 'td-agent.conf.jinja2'
+TD_AGENT_TEMPLATE_PATH = '/etc/td-agent/templates/'
+TD_AGENT_OUTPUT_PATH = '/etc/td-agent/td-agent.conf'
+
+
+def restart_td_agent_process():
+    ''' Restart Fluentd '''
+    process = subprocess.Popen(['service', 'td-agent', 'restart'])
+    exit_code = process.wait(timeout=5)
+    if exit_code:
+        raise Exception("'service td-agent restart' failed with exit code: {0}".format(exit_code))
+
+
+def get_scalyr_api_key():
+    ''' Read Scalyr API key from Taupage config and set in template file '''
+    mainconfig = get_config()
+    config = mainconfig.get('logging')
+    if config:
+        scalyr_api_key = config.get('scalyr_account_key', False)
+    else:
+        scalyr_api_key = False
+
+    if scalyr_api_key:
+        # If scalyr_api_key starts with "aws:kms:" then decrypt key
+        match_kms_key = re.search('aws:kms:', scalyr_api_key, re.IGNORECASE)
+        if match_kms_key:
+            scalyr_api_key = re.sub(r'aws:kms:', '', scalyr_api_key)
+            try:
+                scalyr_api_key = subprocess.check_output(['python3',
+                                                          '/opt/taupage/bin/decrypt-kms.py',
+                                                          scalyr_api_key]).decode('UTF-8').strip()
+            except Exception:
+                logger.error('Failed to run /opt/taupage/bin/decrypt-kms.py')
```

Comment:
Why not just crash here? What's the point of proceeding?

Reply:
There could be addititional destinations whose configuration depends on this script finishing, which should still work if the Scalyr key decryption fails.

Reply:
But then the config will be invalid since you don't quote the api key in the template. If you want to rely on this, please test the behavior.